docs/dev/README.md

```diff
@@ -0,0 +1,110 @@
+
+All modules must have an accompanying doc comment that describes the general functionality of the module. A brief 
+description of the public functions, structs, enums, or other central elements of the module can be included.
+
+### Testing and Linting
+All public and private functions must be appropriately covered by unit tests. Full coverage is intended, which means all
+branches of computation within each function should be thoroughly tested.
+
+In addition, the following commands must not return any warnings or errors for the code currently in master: 
+[cargo build --all-targets](https://doc.rust-lang.org/cargo/commands/cargo-build.html), 
+[cargo clippy --all-targets](https://github.com/rust-lang/rust-clippy), 
+[cargo doc](https://doc.rust-lang.org/cargo/commands/cargo-doc.html), 
+[cargo machete --with-metadata](https://github.com/bnjbvr/cargo-machete), 
+and [cargo test --all-targets](https://doc.rust-lang.org/cargo/commands/cargo-test.html).
+
+### Crates
+To avoid confusion and unnecessary dependencies, a list of crates is included. Note that this only includes crates
+used for purposes such as logging, where multiple crates provide similar functionality.
+
+- Logging - [tracing](https://crates.io/crates/tracing)
+- Async Runtime - [tokio](https://crates.io/crates/tokio)
+- gRPC - [tonic](https://crates.io/crates/tonic)
+- UUID - [uuid](https://crates.io/crates/uuid)
+- SQLite - [rusqlite](https://crates.io/crates/rusqlite)
+- Memory Allocation - [snmalloc-rs](https://crates.io/crates/snmalloc-rs)
+- Hardware Information - [sysinfo](https://crates.io/crates/sysinfo)
+- Property-based Testing - [proptest](https://crates.io/crates/proptest)
+- Temporary Files and Directories - [tempfile](https://crates.io/crates/tempfile)
+
+## Docker
+An environment that includes a local [MinIO](https://min.io/) instance and an edge node using the [MinIO](https://min.io/)
+instance as the remote object store, can be set up using [Docker](https://docs.docker.com/). Note that since
+[Rust](https://www.rust-lang.org/) is a compiled language and a more dynamic `modelardbd` configuration might be needed,
+it is not recommended to use the [Docker](https://docs.docker.com/) environment during active development of `modelardbd`.
+It is however ideal to use during more static deployments or when developing components that utilize `modelardbd`.
+
+Downloading [Docker Desktop](https://docs.docker.com/desktop/) is recommended to make maintenance of the created
+containers easier. Once [Docker](https://docs.docker.com/) is set up, the [MinIO](https://min.io/) instance can be
+created by running the services defined in [docker-compose-minio.yml](docker-compose-minio.yml). The services can
+be built and started using the command:
+
+```shell
+docker-compose -p modelardata-minio -f docker-compose-minio.yml up
+```
+
+After the [MinIO](https://min.io/) service is created, a [MinIO](https://min.io/) client is created to initialize
```

Comment:
Maybe `client is used` as I assume we do not create the client.

Reply:
Fixed.